### PR TITLE
rename scala3doc into scaladoc

### PR DESCRIPTION
**Pull request by romanowski**
Rename scaladoc package to `dotty.tools.scaladoc`

This PR makes a lot of mechanical renames.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -1227,12 +1227,12 @@ object Build {
   val testDocumentationRoot = taskKey[String]("Root directory where tests documentation are stored")
   val generateSelfDocumentation = taskKey[Unit]("Generate example documentation")
   // Note: the two tasks below should be one, but a bug in Tasty prevents that
-  val generateScala3Documentation = inputKey[Unit]("Generate documentation for dotty lib")
+  val generatescaladocumentation = inputKey[Unit]("Generate documentation for dotty lib")
```
Comment (abgruszecki) [inline project/Build.scala:1230]: Intentional?

File: sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala [hunk 2]
```diff
@@ -23,7 +23,7 @@ object DottyPlugin extends AutoPlugin {
     val isDotty = settingKey[Boolean]("Is this project compiled with Dotty?")
     val isDottyJS = settingKey[Boolean]("Is this project compiled with Dotty and Scala.js?")
 
-    val useScala3doc = settingKey[Boolean]("Use Scala3doc as the documentation tool")
+    val useScaladoc = settingKey[Boolean]("Use scaladoc as the documentation tool")
```
Comment (abgruszecki) [inline sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala:26]: Does the new name really make sense here? Perhaps we should call this key `useNewScaladoc`?

Comment (romanowski) [reply sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala:26]: Next step would be removing scala3-doc and this setting as well.

File: project/Build.scala (hunk 1 above)
Comment (romanowski) [reply project/Build.scala]: Fixed.

**Review by BarkingBad — APPROVED** [review]

File: .github/workflows/scaladoc.yaml [hunk 3]
```diff
@@ -64,9 +64,9 @@ jobs:
         with:
           inlineScript: |
             DOC_DEST=pr-${PR_NUMBER:-${GITHUB_REF##*/}}
-            echo uplading docs to https://scala3doc.virtuslab.com/$DOC_DEST
-            az storage container create --name $DOC_DEST --account-name scala3docstorage --public-access container
-            az storage blob sync -s scala3doc/output -c $DOC_DEST --account-name scala3docstorage
+            echo uplading docs to https://scaladoc.virtuslab.com/$DOC_DEST
```
Comment (abgruszecki) [inline .github/workflows/scaladoc.yaml:67]: Is this intentional? The domain doesn't seem available yet...

Comment (romanowski) [reply .github/workflows/scaladoc.yaml]: I've used replace to change scala3doc to scaladoc and in some places (like this one) rename should not be applied

**Pull request merged**